src/main/java/org/oscarehr/PMmodule/dao/ProgramDaoImpl.java

```
@@ -232,214 +296,270 @@ public List<Program> getPrograms() {
     }
 
+    /**
+     * Retrieves programs by gender type.
+     * 
+     * @param genderType the gender type
+     * @return a list of Program objects matching the specified gender type
+     */
     @Override
     public List<Program> getProgramByGenderType(String genderType) {
-        // yeah I know, it's not safe to insert random strings and it's also
-        // inefficient, but unless I fix all the hibernate calls I'm following
-        // convention of
-        // using the hibernate templates and just inserting random strings for now.
-        @SuppressWarnings("unchecked")
-        List<Program> rs = (List<Program>) getHibernateTemplate()
-                .find("FROM Program p WHERE p.manOrWoman = '" + genderType + "'");
-        return rs;
+        TypedQuery<Program> query = entityManager.createQuery(
+            "FROM Program p WHERE p.manOrWoman = ?1", Program.class);
+        query.setParameter(1, genderType);
+        return query.getResultList();
     }
 
+    /**
+     * Saves a program.
+     * 
+     * @param program the Program object to save
+     * @throws IllegalArgumentException if the program is null
+     */
     @Override
     public void saveProgram(Program program) {
         if (program == null) {
             throw new IllegalArgumentException();
         }
         program.setLastUpdateDate(new Date());
 
-        // Adjusting flush mode
-        Session session = currentSession();
-        FlushMode previousFlushMode = session.getHibernateFlushMode();
-        session.setHibernateFlushMode(FlushMode.COMMIT);
-
         try {
-            getHibernateTemplate().saveOrUpdate(program);
+            entityManager.merge(program); // Use merge to save or update
+            if (log.isDebugEnabled()) {
+                log.debug("saveProgram: " + program.getId());
+            }
         } finally {
-            session.setHibernateFlushMode(previousFlushMode); // Restore the original flush mode
+            //swallow
         }
```

open to options but this don't look ideal.

---

I have removed the try-catch-finally here.
We're not supposed to catch and swallow the exception. Instead, we should let it throw the exception so that the higher-level code can handle them.